Make sure imported annotations use the new document metadata

acelaya opened the pull request:
As suggested in https://github.com/hypothesis/client/issues/5743#issuecomment-1695123822, this PR ensures we use the new document title and any other metadata when creating imported annotations.

> Part of https://github.com/hypothesis/client/issues/5743, https://github.com/hypothesis/client/issues/5771 and https://github.com/hypothesis/client/issues/5772

robertknight reviewed (commented):
This looks good, but in the process of reading this I realized that we also need to ensure other document metadata is included, especially the PDF fingerprint. This is necessary to ensure that the URI for the imported annotation is associated with the same fingerprint and other metadata on the backend as if an annotation was created manually.

To reiterate something from Slack, the way the "Import" feature is designed, it should behave the same as if the user manually copied and pasted annotations into a new group. The annotation quote, text and tags are preserved from the import, and then everything else is set the same as if you created a new annotation locally.

acelaya commented:
I have updated the logic to do this:

```ts
{
  // ...
  document: currentFrame?.metadata ?? ann.document,
}
```

instead of

```ts
{
  // ..
  document: {
    title: currentFrame?.metadata.title ?? ann.document.title,
  },
}
```

I have also updated the description referencing all the issues that are related with this change.
